internal/ignition/installmanifests.go

```diff
@@ -548,30 +548,36 @@ func (g *installerGenerator) bootstrapInPlaceIgnitionsCreate(ctx context.Context
 func (g *installerGenerator) expandUserMultiDocYamls(ctx context.Context) error {
 	log := logutil.FromContext(ctx, g.log)
 
-	metadata, err := manifests.GetManifestMetadata(ctx, g.cluster.ID, g.s3Client)
+	manifests, err := manifests.GetClusterManifests(ctx, g.cluster.ID, g.s3Client)
 	if err != nil {
-		return errors.Wrapf(err, "Failed to retrieve manifest matadata")
-	}
-	userManifests, err := manifests.ResolveManifestNamesFromMetadata(
-		manifests.FilterMetadataOnManifestSource(metadata, constants.ManifestSourceUserSupplied),
-	)
-	if err != nil {
-		return errors.Wrapf(err, "Failed to resolve manifest names from metadata")
+		return errors.Wrapf(err, "Failed to retrieve cluster manifests for cluster id %s", g.cluster.ID)
 	}
 
 	// pass a random token to expandMultiDocYaml in order to prevent name
 	// clashes when spliting one file into several ones
 	randomToken := uuid.NewString()[:7]
 
-	for _, manifest := range userManifests {
+	for _, manifest := range manifests {
+
+		// Skip any files that are not user supplied
+		metadata, ok := manifest.Metadata[constants.ManifestSourceAttribute]
+		if ok && metadata != nil && *metadata != constants.ManifestSourceUserSupplied {
+			continue
+		}
+
 		log.Debugf("Looking at expanding manifest file %s", manifest)
 
-		extension := filepath.Ext(manifest)
+		extension := filepath.Ext(manifest.Path)
 		if !(extension == ".yaml" || extension == ".yml") {
 			continue
 		}
 
-		manifestPath := filepath.Join(g.workDir, manifest)
+		pathElements := strings.Split(manifest.Path, "/")
+		if len(pathElements) < 2 {
+			return errors.Wrapf(err, "Failed to normalize manifest path %s into folder and file", manifest.Path)
+		}
+
+		manifestPath := filepath.Join(g.workDir, pathElements[len(pathElements)-2], pathElements[len(pathElements)-1])
```

Comment:
I don't have the full history, but I don't see if it's ok to remove the first folders if there are many
For example "a/b/c/d" will result in "c/d"

Reply:
The input path is always in the format `{cluster_id}/manifests/{folder_name}/{file_name}`
Any other input path is considered to be illegal (and as these directories are created by code, unlikely to vary from this pattern)
If we are worried about this, we could add a check to ensure that len(pathElements) == 4 and this might be better 
WDYT? @pastequo